### DISCUSSION TITLE
[Scheduler] Set default transport for all AsPeriodicTask/AsCronTask attributes

### DISCUSSION THREAD

**Discussion by W0rma** [symfony/symfony#53732] [Q&A] [answered]
The parameter `$transports` of the attributes [`AsCronTask`](https://github.com/symfony/scheduler/blob/7.0/Attribute/AsCronTask.php) and [`AsPeriodicTask`](https://github.com/symfony/scheduler/blob/7.0/Attribute/AsPeriodicTask.php) can be used to define the transport to which the message should be redispatched before being further redispatched to its corresponding handler.

Docs: https://symfony.com/doc/current/scheduler.html#efficient-management-with-symfony-scheduler

In my project I want to redispatch every task to the same transport.

Do I need to define `$transports` for every single task or is there a way to define a default transport?

**Comment by MatTheCat** [comment, marked as answer]
Did you try extending these attributes with a static transport and use them on your tasks?

**Reply by W0rma** [reply]
Yes, I'm aware of that solution.

I just wanted to clarify if there is some built-in solution for that which doesn't require extending the attribute.

**Reply by MatTheCat** [reply]
Don't know how “built-in” this would be, but you could register a compiler pass after the `AddScheduleMessengerPass` to change the transportsʼ definition.

But if youʼre gonna change Symfonyʼs default it sounds better to make it explicit, using your own attributes.
